docs/core/v3.0.rst

```diff
@@ -0,0 +1,1761 @@
+    as defined in this specification, ignoring any name/value pairs
+    where the name is not defined here. See also the `stability policy`_.
+
+    Note that this value is given as a URI rather than as a simple
+    version number string to help with discovery of this
+    specification.
+
+``metadata_encoding``
+^^^^^^^^^^^^^^^^^^^^^
+
+    A string containing the URI pointing to a document describing the method
+    used for encoding group and array metadata documents.
+
+    For document using the default JSON encoding and format describe in this document
+    then the value must be ``"https://purl.org/zarr/spec/core/3.0``.
+
+``metadata_key_suffix``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+    A string containing a suffix to add to the metadata keys when saving into
+    the store. By default ``".json"``.
+
+    .. note::
+
+      This suffix is used to allow non hierarchy
+      browsing and editing by non-zarr-aware tools.
+
+``extensions``
+^^^^^^^^^^^^^^
+
+    An array containing zero or more objects, each of which identifies
+    an extension and provides any additional extension
+    configuration metadata. Each object must contain the name
+    ``extension`` whose value is a URI that identifies a Zarr
+    extension and dereferences to a human readable representation of
+    the extension specification. Each object must also contain the
+    name ``must_understand`` whose value is either the literal
+    ``true`` or ``false``. Each object may also contain the name
+    ``configuration`` whose value is defined by the
+    extension.
+
+    If an implementation of this specification encounters an extension
+    that it does not recognize, but the value of ``must_understand``
+    is ``false``, then the extension may be ignored and processing may
+    continue. If the extension is not recognized and the value of
+    ``must_understand`` is ``true`` then processing must terminate and
+    an appropriate error raised.
+
+For example, below is an entry point metadata document, specifying that
+JSON is being used for encoding of group and array metadata::
+
+    {
+        "zarr_format": "https://purl.org/zarr/spec/core/3.0",
+        "metadata_encoding": "https://purl.org/zarr/spec/core/3.0",
+        "metadata_key_suffix" : ".json",
+        "extensions": []
+    }
+
+For example, below is an entry point metadata document as above, but also
+specifying that an extension is being used which may be
+ignored if not understood::
+
+    {
+        "zarr_format": "https://purl.org/zarr/spec/core/3.0",
+        "metadata_encoding": "https://purl.org/zarr/spec/core/3.0",
+        "metadata_key_suffix" : ".json",
+        "extensions": [
+            {
+                "extension": "http://example.org/zarr/extension/foo",
+                "must_understand": false,
+                "configuration": {
+                    "foo": "bar"
+                }
+            }
+        ]
+    }
+
```

Comment:
In addition to root-level extensions, this would be a place to add root-level storage transformers.